src/assets/wise5/authoringTool/edit-branch/edit-branch.component.ts

```diff
@@ -208,8 +213,51 @@ export class EditBranchComponent extends AbstractBranchAuthoringComponent {
   }
 
   protected async submit(): Promise<void> {
-    alert(JSON.stringify(this.getBranchParams(), null, 2));
-    // TODO: update branch
-    // this.router.navigate(['..'], { relativeTo: this.route });
+    this.submitting = true;
+    const params = this.getBranchParams();
+    this.updateTransitions(params);
+    this.updateTransitionLogic(params);
+    this.saveProject();
+  }
+
+  private updateTransitions(params: any): void {
+    for (let x = 0; x < this.node.transitionLogic.transitions.length; x++) {
+      const transition = this.node.transitionLogic.transitions[x];
+      if (params.criteria === this.SCORE_VALUE) {
+        transition.criteria = [
+          {
+            name: this.SCORE_VALUE,
+            params: {
+              componentId: params.componentId,
+              nodeId: params.nodeId,
+              scores: [params.paths[x]]
+            }
+          }
+        ];
+      } else if (params.criteria === this.CHOICE_CHOSEN_VALUE) {
+        transition.criteria = [
+          {
+            name: this.CHOICE_CHOSEN_VALUE,
+            params: {
+              choiceIds: [params.paths[x]],
+              componentId: params.componentId,
+              nodeId: params.nodeId
+            }
+          }
+        ];
+      } else {
+        delete transition.criteria;
+      }
+    }
+  }
+
+  private updateTransitionLogic(params: any): void {
+    if (params.criteria === this.WORKGROUP_ID_VALUE || params.criteria === this.RANDOM_VALUE) {
+      this.node.transitionLogic.howToChooseAmongAvailablePaths = params.criteria;
+      this.node.transitionLogic.whenToChoosePath = 'enterNode';
+    } else {
+      this.node.transitionLogic.howToChooseAmongAvailablePaths = this.RANDOM_VALUE;
```

Comment:
If we choose "choice chosen" in the UI, why do we set the field to "random" in the JSON here?

Reply:
If multiple paths are satisfied, we choose one of them randomly. For example, if one of the paths is set to a score of 1 and another path is set to a score of 1, we will randomly choose between those two paths. This is how it has always been. I don't know if this logic is ever utilized and we could probably change it to just choose the first path that is satisfied if we wanted in the future.

Reply:
I see. We should change this in the future so that the field says "choiceChosen" instead.